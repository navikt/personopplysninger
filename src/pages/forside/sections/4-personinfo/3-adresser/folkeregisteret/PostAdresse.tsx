import React from "react";
import { Normaltekst } from "nav-frontend-typografi";
import AdressePanel from "../komponenter/AdressePanel";
import GateAdresse from "../komponenter/GateAdresse";
import Postnummer from "../komponenter/Postnummer";
import { Postadresse } from "../../../../../../types/adresser/postadresse";

interface Props {
  postadresse: Postadresse;
}

const PostAdresse = (props: Props) => {
  const {
    adresse1,
    adresse2,
    adresse3,
    postnummer,
    poststed,
    land
  } = props.postadresse;
  return (
    <AdressePanel tittel="adresse.postadresse">
      <>
        <GateAdresse
          adresse1={adresse1}
          adresse2={adresse2}
          adresse3={adresse3}
        />
        <Postnummer postnummer={postnummer} poststed={poststed} />
        {land && (
          <div className="adresse__linje">
            <Normaltekst>{land}</Normaltekst>
          </div>
        )}
      </>
    </AdressePanel>
  );
};

export default PostAdresse;
