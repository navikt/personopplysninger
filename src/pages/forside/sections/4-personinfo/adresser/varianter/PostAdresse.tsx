import React from "react";
import { Normaltekst } from "nav-frontend-typografi";
import AdressePanel from "../../../../../../components/adresse/AdressePanel";
import GateAdresse from "../../../../../../components/adresse/GateAdresse";
import Postnummer from "../../../../../../components/adresse/Postnummer";
import { Postadresse } from "../../../../../../types/adresser/postadresse";
import Kilde from "../../../../../../components/kilde/Kilde";

interface Props {
  postadresse: Postadresse;
}

const PostAdresse = (props: Props) => {
  const {
    adresse1,
    adresse2,
    adresse3,
    postnummer,
    poststed,
    land
  } = props.postadresse;
  return (
    <AdressePanel tittel="adresse.postadresse">
      <>
        <GateAdresse
          adresse1={adresse1}
          adresse2={adresse2}
          adresse3={adresse3}
        />
        <Postnummer postnummer={postnummer} poststed={poststed} />
        {land && (
          <div className="adresse__linje">
            <Normaltekst>{land}</Normaltekst>
          </div>
        )}
        <Kilde
          kilde="personalia.source.folkeregisteret"
          lenke="https://www.skatteetaten.no/person/folkeregister/"
          lenkeTekst="personalia.link.folkeregisteret"
        />
      </>
    </AdressePanel>
  );
};

export default PostAdresse;
