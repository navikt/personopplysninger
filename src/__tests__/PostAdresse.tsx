import * as React from "react";
import ReactTestRenderer from "react-test-renderer";
import wrapIntl from "./__utils__/IntlTestHelper";
import PostAdresse from "../pages/forside/sections/4-personinfo/3-adresser/folkeregisteret/PostAdresse";

test("render Postadresse with content", () => {
  const postadresse = {
    adresse1: "12345 HFJEF IQUFJ HFURHGHQBEUQ",
    adresse2: "BATOOM LUNCHAGRAMA,GONGRUPEA,DD",
    adresse3: "GGLAM,10000 HCAMGHU,THAILAND",
    datoFraOgMed: "2018-03-13",
    land: "THA",
    postnummer: "9913"
  };

  const component = ReactTestRenderer.create(
    wrapIntl(<PostAdresse postadresse={postadresse} />)
  );
  expect(component.toJSON()).toMatchSnapshot();
});
