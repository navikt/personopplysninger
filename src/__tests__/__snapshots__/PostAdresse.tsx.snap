// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Postadresse with content 1`] = `
<div
  className="adresse__box"
>
  <hr
    className="box__linje-bred"
  />
  <div
    className="adresse__tittel"
  >
    <h2
      className="typo-undertittel"
    >
      <span>
        Postadresse
      </span>
    </h2>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      12345 HFJEF IQUFJ HFURHGHQBEUQ
    </p>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      BATOOM LUNCHAGRAMA,GONGRUPEA,DD
    </p>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      GGLAM,10000 HCAMGHU,THAILAND
    </p>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      9913

    </p>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      THA
    </p>
  </div>
  <div
    className="kilde__container"
  >
    <div
      className="kilde__seksjon"
    >
      <p
        className="typo-undertekst"
      >
        <span>
          Kilde: Folkeregisteret
        </span>
      </p>
    </div>
    <div
      className="kilde__seksjon"
    >
      <p
        className="typo-normal"
      >
        <a
          className="lenke"
          href="https://www.skatteetaten.no/person/folkeregister/"
        >
          <span>
            Endre i Folkeregisteret
          </span>
        </a>
      </p>
    </div>
  </div>
</div>
`;
